tests/e2e/specs/Modeler.spec.js

```diff
@@ -0,0 +1,216 @@
+  <bpmn:process id="Process_1" isExecutable="true">
+    <bpmn:startEvent id="node_2" name="${nodeName}" />
+  </bpmn:process>
+  <bpmndi:BPMNDiagram id="BPMNDiagram_1">
+    <bpmndi:BPMNPlane id="BPMNPlane_1" bpmnElement="Process_1">
+      <bpmndi:BPMNShape id="node_2_di" bpmnElement="node_2">
+        <dc:Bounds x="150" y="150" width="36" height="36" />
+      </bpmndi:BPMNShape>
+    </bpmndi:BPMNPlane>
+  </bpmndi:BPMNDiagram>
+</bpmn:definitions>`;
+};
+
+const dragFromSourceToDest = (source, dest, position) => {
+  const dataTransfer = new DataTransfer();
+  cy.get(source).trigger('dragstart', { dataTransfer });
+  cy.get(dest).trigger('dragenter');
+  cy.get(dest).trigger('drop', { offsetX: position.x, offsetY: position.y });
+};
+
+function coordinateGenerator() {
+  return Math.floor(Math.random() * 500) + 150;
+}
+
+function typeIntoTextInput(selector, value) {
+  cy.wait(500);
+  cy.get(selector).focus().clear().type(value, {force: true});
+  cy.wait(500);
+}
+
+function connectNode(source, postionX, positionY) {
+  cy.get(source).click().trigger('mousemove', { x: postionX, y: positionY, force: true});
+}
+
+const nodeTypes = [
+  '.processmaker-modeler-task',
+  '.processmaker-modeler-end-event',
+  '.processmaker-modeler-script-task',
+  '.processmaker-modeler-exclusive-gateway',
+  '.processmaker-modeler-text-annotation',
+  '.processmaker-modeler-pool',
+];
+
+describe('Modeler', () => {
+  beforeEach(()=> {
+    cy.visit('/');
+    cy.reload();
+  });
+
+  it('Renders the application without errors', () => {
+    cy.get('.navbar').should('contain', 'ProcessMaker Modeler');
+  });
+
+  it('Renders list of nodes', () => {
+    cy.get('.modeler').children().should('have.length', 2);
+
+    nodeTypes.forEach(type => {
+      dragFromSourceToDest(
+        type,
+        '.paper-container',
+        { x: coordinateGenerator(), y: coordinateGenerator()},
+      );
+    });
+    cy.get('.modeler').children().should('have.length', 8);
+  });
+
+  it('Creates top and bottom lane', () => {
+    const poolSelector = '#v-24';
+    const topLaneSeletor = '#v-26';
+    const bottomLaneSeletor = '#v-29';
+
+    dragFromSourceToDest(
+      '.processmaker-modeler-pool',
+      '.paper-container',
+      { x: 100, y: 100},
+    );
+    cy.get(poolSelector).click();
+    cy.get(topLaneSeletor).click({ force: true });
+    cy.get(bottomLaneSeletor).click({ force: true });
+
+  });
+
+  it('Update task name', ()=> {
+    dragFromSourceToDest(
+      '.processmaker-modeler-task',
+      '.paper-container',
+      { x: coordinateGenerator(), y: coordinateGenerator()},
+    );
+
+    cy.get('.joint-viewport').find('.joint-type-processmaker-components-nodes-task').click({force: true});
+    typeIntoTextInput('[name=\'name\']', 'testing');
+    cy.get('[name=\'name\']').should('have.value', 'testing');
+  });
+
+  it('Update exclusive gateway name', ()=> {
```

Comment:
In some places it's `()=>` and in others it's `() =>` (space between `()` and `=>` in arrow functions). I think this should match the style in the rest of the app and include the space. This stuff should go into our ESLint config though (thought it was already there) so we don't have to think about it.